go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -1055,6 +1131,32 @@ func (vs *vstreamer) extractRowAndFilter(plan *streamerPlan, data []byte, dataCo
 	return ok, filtered, partial, err
 }
 
+// addEnumAndSetMappingstoPlan sets up any necessary ENUM and SET integer to string mappings.
+func addEnumAndSetMappingstoPlan(plan *Plan, cols []*querypb.Field, metadata []uint16) error {
+	plan.EnumSetValuesMap = make(map[int]map[int]string)
+	for i, col := range cols {
+		// If the column is a CHAR based type with a binary collation (e.g. utf8mb4_bin) then
+		// the actual column type is included in the second byte of the event metadata while
+		// the event's type for the field is BINARY. This is true for ENUM and SET types.
+		var mysqlType uint16
+		if sqltypes.IsQuoted(col.Type) {
+			mysqlType = metadata[i] >> 8
+		}
+		if col.Type == querypb.Type_ENUM || mysqlType == mysqlbinlog.TypeEnum ||
+			col.Type == querypb.Type_SET || mysqlType == mysqlbinlog.TypeSet {
+			// Strip the enum() / set() parts out.
```

Comment:
We should use `schema.ParseEnumValues()` and `schema.ParseSetValues()` here.

Reply:
That's exactly what I'm doing two lines below. None of these functions accept e.g. `enum('val1, val2')`, they only accept `'val1,val2'` which is why I'm removing the enum() / set() parts.

Reply:
Let's keep it this way for the time being. I have it on my TODO list to have a proper `enum` parser, and I'm setting a mental bookmark to revisit this code once that's done.